src/Resource/Collection.js

```diff
@@ -4,32 +4,42 @@ import ParameterBag from './ParameterBag';
 /**
  * @param {String|Object} name
  * @param {HyralRepository} repository
- * @param {Object} serializedData
+ * @param {Object} dataDefaults
+ * @param {ParameterBag|Object} parameterBag
  */
-function Collection(name, repository, serializedData = {
+function Collection(
   name,
-  items: [],
-  metadata: {
-    loaded: false,
-    parameterBagData: null,
-    paging: {
-      count: null,
-      pages: null,
+  repository,
+  dataDefaults = {
+    items: [],
+    metadata: {
+      loaded: false,
+      paging: {
+        count: null,
+        pages: null,
+      },
     },
   },
-}) {
+  parameterBag = null,
+) {
   if (!new.target) {
     throw Error('Collection() must be called with new');
   }
-  const serializebleData = cloneDeep(serializedData);
+
+  const data = cloneDeep(dataDefaults);
   const privateData = {
     loading: false,
     lastParameterBagState: null,
-    parameterBag: serializebleData.metadata.parameterBagData
-      ? ParameterBag.createFromData(serializebleData.metadata.parameterBagData)
-      : null,
+    parameterBag: null,
   };
 
+  if (parameterBag instanceof ParameterBag) {
```

Comment:
Als we dit gaan gebruiken moeten we er zeker van zijn dat dit in dit geval altijd werkt.
instanceof in javascript is niet in alle gevallen safe voor zover ik weet.